packages/CodingStandard/src/Sniffs/Architecture/ExplicitExceptionSniff.php

```diff
@@ -0,0 +1,82 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Sniffs\Architecture;
+
+use BadFunctionCallException;
+use BadMethodCallException;
+use DomainException;
+use Exception;
+use InvalidArgumentException;
+use LengthException;
+use LogicException;
+use OutOfBoundsException;
+use OutOfRangeException;
+use OverflowException;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use RangeException;
+use RuntimeException;
+use Throwable;
+use UnderflowException;
+use UnexpectedValueException;
+
+final class ExplicitExceptionSniff implements Sniff
+{
+    /**
+     * @return int[]
+     */
+    public function register(): array
+    {
+        return [T_THROW];
+    }
+
+    /**
+     * @param int $position
+     */
+    public function process(File $file, $position): void
+    {
+        // check "throw new" construction
+        $newPosition = $file->findNext(T_NEW, $position, $position + 3);
+        if (! $newPosition) {
+            return;
+        }
+
+        $exceptionNamePosition = $file->findNext([T_STRING], $position);
+        $exceptionNameToken = $file->getTokens()[$exceptionNamePosition];
+
+        $exceptionName = $exceptionNameToken['content'];
+
+        if (! $this->isNativeExceptionName($exceptionName)) {
+            return;
+        }
+
+        $file->addError(sprintf(
+            'Use explicit and informative exception names over generic ones like "%s".',
+            $exceptionName
+        ), $position, self::class);
+    }
+
+    /**
+     * Check against official list: http://php.net/manual/en/spl.exceptions.php
```

Comment:
Give the official list of `PHP Exceptions` will definitely help devs to know why the sniff catched it.